--- pd-0.41-4/src/configure.in	2008-03-14 17:03:00.000000000 -0700
+++ pd-0.41-4/src/configure.in.new	2008-12-03 11:56:04.000000000 -0800
@@ -148,6 +148,7 @@
 #    exit -1
 fi
 
+AC_CHECK_LIB(tcl8.6, main,,
 AC_CHECK_LIB(tcl85, main,,
         AC_CHECK_LIB(tcl8.5, main,,
         AC_CHECK_LIB(tcl84, main,,
@@ -155,8 +156,9 @@
                 AC_CHECK_LIB(tcl8.3, main,,
                     AC_CHECK_LIB(tcl8.2, main,,
                                     AC_CHECK_LIB(tcl8.0, main,,
-                            echo no tcl library found; exit 1)))))))
+                            echo no tcl library found; exit 1))))))))
 
+AC_CHECK_LIB(tk8.6, main,,
 AC_CHECK_LIB(tk85, main,,
         AC_CHECK_LIB(tk8.5, main,,
        AC_CHECK_LIB(tk84, main,,
@@ -164,7 +166,7 @@
                 AC_CHECK_LIB(tk8.3, main,,
                     AC_CHECK_LIB(tk8.2, main,,
                                     AC_CHECK_LIB(tk8.0, main,,
-                            echo no tk library found; exit 1)))))))
+                            echo no tk library found; exit 1))))))))
 
 
 if test x$tk != "xno"; then
--- pd-0.41-4/src/t_tkcmd.c	2008-03-14 17:03:00.000000000 -0700
+++ pd-0.41-4/src/t_tkcmd.c.new	2008-12-03 12:19:52.000000000 -0800
@@ -2,6 +2,8 @@
 * For information on usage and redistribution, and for a DISCLAIMER OF ALL
 * WARRANTIES, see the file, "LICENSE.txt," in this distribution.  */
 
+#define USE_INTERP_RESULT
+
 #ifndef MSW     /* in unix this only works first; in MSW it only works last. */
 #include "tk.h"
 #endif
@@ -541,7 +543,7 @@
     result = Tcl_Eval(tk_pdinterp,  s);
     if (result != TCL_OK)
     {
-        if (*tk_pdinterp->result) printf("%s\n",  tk_pdinterp->result);
+        if (*Tcl_GetStringResult(tk_pdinterp)) printf("%s\n",  Tcl_GetStringResult(tk_pdinterp));
     }
 }
 
